_includes/scripts.njk

```diff
@@ -40,3 +40,229 @@
+        "always-months": "bool",
+        clickable: "bool",
+        datetime: "string",
+        raw: "bool",
+      };
+
+      constructor() {
+        super();
+        this.attachShadow({ mode: 'open' });
+        this._timer = null;
+      }
+
+      connectedCallback() {
+        this._render();
+      }
+
+      disconnectedCallback() {
+        this._clearTimer();
+      }
+
+      attributeChangedCallback(name, oldVal, newVal) {
+        if ((name === 'datetime' || name === 'raw' || name === 'abbreviate-months' || name === 'always-months') && oldVal !== newVal) {
+          this._render();
+        }
+        if (name === 'clickable') {
+          if (newVal !== null) {
+            this.addEventListener('click', this._toggleRaw);
+          } else {
+            this.removeEventListener('click', this._toggleRaw);
+          }
+        }
+      }
+
+      _toggleRaw = (ev) => {
+        if (ev.altKey) {
+          return; // alt/option + click to avoid toggle
+        }
+        const sel = typeof this.shadowRoot.getSelection === "function"
+          ? this.shadowRoot.getSelection()
+          : document.getSelection();
+        if (sel && sel.rangeCount && !sel.isCollapsed) {
+          return;
+        }
+        this.toggleAttribute("raw");
+      };
+
+      _clearTimer() {
+        if (this._timer) {
+          clearTimeout(this._timer);
+          this._timer = null;
+        }
+      }
+
+      _parseDate(raw) {
+        if (!raw || typeof raw !== 'string') {
+          return null;
+        }
+        const normalized = raw.replace(/\s+/, 'T');
+        const d = new Date(normalized);
+        if (!isNaN(d.getTime())) {
+          return d;
+        }
+        const d2 = new Date(raw);
+        return isNaN(d2.getTime()) ? null : d2;
+      }
+
+      _format(now, dt) {
+        const ms = now - dt;
+        const minute = 60 * 1000;
+        const hour = 60 * minute;
+        const day = 24 * hour;
+        const year = 365 * day;
+
+        const todayStart = new Date(now.getFullYear(), now.getMonth(), now.getDate());
+        const dtStart = new Date(dt.getFullYear(), dt.getMonth(), dt.getDate());
+        const yesterdayStart = new Date(todayStart.getTime() - day);
+
+        // Always show month if attribute is present
+        const alwaysMonths = this.hasAttribute('always-months');
+        // Use full month if attribute is present, otherwise short (Jan vs January)
+        const abbreviate = this.hasAttribute('abbreviate-months');
+        const monthFormat = abbreviate ? { month: 'short' } : { month: 'long' };
+        const monthFmt = new Intl.DateTimeFormat('en-US', monthFormat);
+        const yearNum = new Intl.DateTimeFormat('en-US', { year: 'numeric' });
+        const hm = new Intl.DateTimeFormat('en-US', { hour: '2-digit', minute: '2-digit', hour12: false });
+
+        if (!alwaysMonths && ms > 7 * year) {
+          return yearNum.format(dt);
+        }
+        if (ms > 1 * year) {
```

Comment:
Just cause I'm curious, is this logic based on a reference implementation?

Reply:
It is simplified a lot.  Typically, after yesterday, you'd switch to "Mon 12:11" for 5 or so days.  Then "April 1, 2025" if the year is this year...  The last switch to "2017" is also arbitrary late because we're usually have not that many updates and I didn't want to switch too early (=too often).